packages/angular/src/generators/init/init.ts

```diff
@@ -51,6 +52,15 @@ export async function angularInitGenerator(
   const options = normalizeOptions(rawOptions);
   setDefaults(tree, options);
 
+  const packagesToInstall = [
+    '@angular-devkit/core',
+    '@angular-devkit/schematics',
+    '@schematics/angular',
+  ];
+  packagesToInstall.forEach((packageToInstall) =>
+    ensurePackage(tree, packageToInstall, angularVersion)
```

Comment:
We can't just use `ensurePackage` as is here because it will overwrite those packages' versions if they exist on a lower version. That's not what we need here for a couple of reasons:

- the `init` generator doesn't run only at workspace creation, it runs every time an app or lib is created, so we don't have guarantees that it is running only for the first time
- we now support multiple versions of Angular (v14 and v15), so a workspace can be in a supported lower version of Angular and those packages' versions would be overwritten by the `angularVersion` we have in our constants which is always the latest

So, in this use case, we need only to install them if they are not already installed, regardless of the version they are installed. To be more accurate, if they are installed, we need to validate they are in the right range we support and throw otherwise. If they are not installed, but `@angular-devkit/build-angular` is, we should add them with the version `@angular-devkit/build-angular` is installed.

This is exactly why I preferred not to do this and handle it in the two places we do need to handle it: creating a new workspace and migrating from Angular CLI. The only other scenario that is not handled by those two is when manually adding the `@nrwl/angular` plugin to a workspace and either run the `init` generator or directly generate a lib or app. During the installation of `@nrwl/angular` the users will get warnings (or errors depending on the pm) regarding missing peer deps.

I'm happy to sync about this matter offline if needed.